include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -0,0 +1,506 @@
+                                    const auto& __wg_aggregate = __partials_acc[__i];
+                                    const auto& __b_seg_end = __seg_ends_acc[__i];
+
+                                    if (__first)
+                                    {
+                                        __agg_collector = __wg_aggregate;
+                                        __first = false;
+                                    }
+                                    else
+                                        __agg_collector = __binary_op(__wg_aggregate, __agg_collector);
+
+                                    // current aggregate is the last aggregate
+                                    if (__b_seg_end)
+                                        break;
+                                }
+                            }
+                        }
+
+                        __ag_exists = sycl::group_broadcast(__group, __ag_exists);
+                        if (!__ag_exists)
+                            return;
+
+                        __agg_collector = sycl::group_broadcast(__group, __agg_collector);
+
+                        // 2c. Second pass over the keys, reidentifying end segments and applying work group
+                        // aggregates if appropriate.
+                        ::std::size_t __local_min_key_idx = __n - 1;
+
+                        // find the smallest segment end index in the work item
+                        for (int32_t __i = __start; __i < __end; ++__i)
+                        {
+                            if (__i == __n - 1 || __keys[__i] != __keys[__i + 1])
+                            {
+                                if (__i < __local_min_key_idx)
+                                    __local_min_key_idx = __i;
+                            }
+                        }
+
+                        ::std::size_t __wg_min_seg_end =
+                            sycl::reduce_over_group(__group, __local_min_key_idx, sycl::minimum<>());
+
+                        // the first group will never have an aggregate to apply
+                        if (__group_id == 0)
+                            return;
+                        
+                        // apply work group aggregates
+                        for (int32_t __i = __start; __i < sycl::minimum{}(__wg_min_seg_end + 1, __end); ++__i)
+                            __out_values[__i] = __binary_op(__agg_collector, __out_values[__i]);
+                    });
+            })
+            .wait();
+    }
+
+    template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3,
+              typename _BinaryPredicate, typename _BinaryOperator, typename _T>
+    void
+    sycl_exclusive_scan_by_segment(_ExecutionPolicy&& __exec, _Range1&& __keys, _Range2&& __values,
+                                   _Range3&& __out_values, _BinaryPredicate __binary_pred, _BinaryOperator __binary_op,
+                                   _T __init = _T{})
```

Comment:
`_T __init = _T{}` adds additional requirement for type` _T `. Can we avoid that?

Reply:
Yes, I will change this.